### PR TITLE
refactor(jina): add type checking by monkeytype

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
Changes applied automatically by using `MonkeyType` adding Type Hints. Credit to @deepankarm and @bwanglzu for the directions. It is really easy to make `MonkeyType` work

Usage example:

```bash
pip install MonkeyType
pip install pytest-monkeytype 
pytest --monkeytype-output=./monkeytype.sqlite3 tests/unit/drivers/test_segmenter_driver.py -v -s
# It will list the modules affected by the run
monkeytype list-modules
# Apply changes to the modules u choose
monkeytype apply jina.executors.crafters
```

**Take home message**
Next time work on a smaller set of files because it needs some supervision

### CODE REVIEW

File: jina/helper.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def _replace(sub_d: Union[Dict, List], p):
     os.system('color')
 
 
-def colored(text, color=None, on_color=None, attrs=None):
+def colored(text: str, color: Optional[str]=None, on_color: None=None, attrs: None=None) -> str:
```
Comment (JoanFM) [inline jina/helper.py:425]: I guess it needs to see more examples to see more options

File: jina/helper.py [hunk 2]
```diff
@@ -178,18 +181,14 @@ def parse_arg(v: str):
     return v
 
 
-def countdown(t: int, logger=None, reason: str = 'I am blocking this thread'):
-    if not logger:
-        sys.stdout.write('\n')
-        sys.stdout.flush()
+def countdown(t: int, reason: str = 'I am blocking this thread') -> None:
```
Comment (JoanFM) [inline jina/helper.py:184]: thanks to MonkeyType I saw that logger is always None

File: jina/counter.py [hunk 3]
```diff
@@ -7,21 +7,21 @@
 
 
 class BaseCounter(Iterator):
-    def __init__(self, seed: int = 0):
+    def __init__(self, seed: int = 0) -> None:
```
Comment (bwanglzu) [inline jina/counter.py:10]: it's interesting to see monkeytype add `None` to a constructor. normally we never add return type hints to `__init__`, maybe there's a way to disable/ignore it?

Comment (JoanFM) [reply jina/counter.py:10]: I will remove myself for the time being yes.

File: jina/helper.py [hunk 4]
```diff
@@ -9,12 +9,15 @@
...
-from typing import Iterator, Any, Union, List, Dict
+from typing import Tuple, Optional, Iterator, Any, Union, List, Dict, Set, TextIO
 
 import numpy as np
 from ruamel.yaml import YAML, nodes
 
 from . import JINA_GLOBAL
+from argparse import ArgumentParser, Namespace
+from io import StringIO
+from uvloop import Loop
```
Comment (bwanglzu) [inline jina/helper.py:20]: one suggestion, in python we normally follow the import sequence of 

```python
# First import python lib dependencies
import io
import time

# Then import 3rd party dependencies
import numpy as np
import pandas as pd

# Last import internal methods and classes
from jina.xxx import SomeRandomClass
```

File: jina/counter.py (hunk 3 above)
Comment (deepankarm) [reply jina/counter.py:10]: I think, the reason `MonkeyType` adds this is - [PEP-0484](https://www.python.org/dev/peps/pep-0484/#the-meaning-of-annotations) says, `__init__` should be annotated with `-> None` [Stackoverflow reference](https://stackoverflow.com/a/46779290/3925392)

Comment (bwanglzu) [reply jina/counter.py:10]: ah, there's a reason behind it , haha.  Previously we add type hints manually, and we just ignore:
1. Magic methods such as `__enter__`, `__call__`.
2. Private methods started with `_`.
But it's just my personal habbit

Comment (JoanFM) [reply jina/counter.py]: Oh I just removed them. Should I put them back?

Comment (JoanFM) [reply jina/counter.py]: I would vote to remove `__init__` . `__enter__` and `__call__` are also good candidates to remove, but at least there is a type.

For "private" methods I vote to add them.

Comment (bwanglzu) [reply jina/counter.py]: take easy, we can have a meeting at some point and clarify everything :)

File: tests/unit/drivers/test_concat_driver.py [hunk 5]
```diff
@@ -30,12 +30,12 @@ class ConcatDriverTestCase(JinaTestCase):
...
     def test_direct_concat(self):
         doc1, doc2 = input_fn()
-
         t1 = np.concatenate([pb2array(doc1.embedding), pb2array(doc2.embedding)], axis=0)
         doc1.embedding.buffer += doc2.embedding.buffer
         doc1.embedding.shape[0] += doc2.embedding.shape[0]
         t2 = pb2array(doc1.embedding)
-
+        self.assertEqual(t1.shape[0], 10)
+        self.assertEqual(t2.shape[0], 10)
```
Comment (bwanglzu) [inline tests/unit/drivers/test_concat_driver.py:38]: can we slowly replace these unittest style assert into pytest style?
into:

```python
assert t1.shape[0] == 10
```

Comment (JoanFM) [reply tests/unit/drivers/test_concat_driver.py:38]: let's do in a separate PR, I am checking why this test is failing. In my local environment it works.

File: tests/unit/drivers/test_concat_driver.py [hunk 6]
```diff
@@ -9,16 +9,16 @@
...
-    c.embedding.CopyFrom(array2pb(np.random.random([5]).astype('float32')))
+    c.embedding.CopyFrom(array2pb(np.random.random([5])))
     doc2 = Document()
     doc2.id = 2
-    doc2.embedding.CopyFrom(array2pb(np.random.random([3]).astype('float32')))
+    doc2.embedding.CopyFrom(array2pb(np.random.random([3])))
     d = doc2.chunks.add()
     d.id = 4
-    d.embedding.CopyFrom(array2pb(np.random.random([9]).astype('float32')))
+    d.embedding.CopyFrom(array2pb(np.random.random([9])))
```
Comment (bwanglzu) [inline tests/unit/drivers/test_concat_driver.py]: a perfect place to use parameterized test :)

Comment (JoanFM) [reply tests/unit/drivers/test_concat_driver.py]: can u develop this idea in a small example PR maybe? No rush

Comment (bwanglzu) [reply tests/unit/drivers/test_concat_driver.py]: sure

**Pull request merged**